src/NHibernate/Cfg/Environment.cs

```diff
@@ -208,6 +208,14 @@ public static string Version
 		/// </summary>
 		public const string OdbcDateTimeScale = "odbc.explicit_datetime_scale";
 
+		// Obsolete Since v5.0
+		/// <summary>
+		/// Disable switching built-in NHibernate date-time types from DbType.DateTime to DbType.DateTime2
+		/// for dialects supporting datetime2.
+		/// </summary>
+		[Obsolete("Migrate SQL datetime type to datetime2 instead.")]
+		public const string SqlTypesKeepDateTime = "sql_types.keep_datetime";
```

Comment:
`DateTime2` being unsupported by OdbcDriver, we may use it for allowing using 2008 dialect with Odbc. So I think I will remove the obsolete attribute.

The subject is a bit deeper since Odbc does not understand `Time` either, and likely `Date` too: we should have a mean to switch those types too. So we can also decide that only dialects up to 2005 are compatible with ODBC.
(By the way this [page](https://docs.microsoft.com/en-us/sql/relational-databases/native-client-odbc-date-time/data-type-support-for-odbc-date-and-time-improvements) states ODBC does support SQL2008 types, but how are we supposed to do in a .Net context, mystery.)

An other option would be to try moving the type override from dialect to driver. But currently `IMapping` interface which is where I pull the dialect from for refining types is not much suitable for having the driver, unless we add the ability for `Configuration.Mapping` to instantiate a driver.

I think I will just remove the `Obsolete` attribute.